pkg/influx/run.go

```diff
@@ -0,0 +1,50 @@
+package influx
+
+import (
+	"flag"
+	"os"
+
+	"github.com/cortexproject/cortex/pkg/util/fakeauth"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/grafana/dskit/flagext"
+	"github.com/weaveworks/common/logging"
+	"github.com/weaveworks/common/server"
+)
+
+func Run() error {
+	var (
+		serverConfig server.Config
+		enableAuth   bool
+		apiConfig    APIConfig
+	)
+
+	// Register flags.
+	flag.BoolVar(&enableAuth, "auth.enable", true, "enable X-Scope-OrgId header")
```

Comment:
@leizor 

As a rule, libraries should never define flags, only binaries should do that.  Instead, all setup should be done programmatically, possibly with a config struct.  the outermost binary can define flags as needed.  Can you create a followup PR that moves the flags out of the library and into the binary?

Reply:
I approved this but also Agree with @ywwg 's comment. 

You can mirror how other projects like query-tee https://github.com/cortexproject/cortex/blob/master/cmd/query-tee/main.go#L17-L31 does it.